src/Nethermind/Nethermind.Facade/Eth/RpcTransaction/TransactionForRpc.cs

```diff
@@ -61,13 +65,16 @@ public virtual Transaction ToTransaction()
     public abstract void EnsureDefaults(long? gasCap);
     public abstract bool ShouldSetBaseFee();
 
-    internal class JsonConverter : JsonConverter<TransactionForRpc>
+    internal class TransactionJsonConverter : JsonConverter<TransactionForRpc>
     {
-        // NOTE: Should we default to a specific TxType?
-        private const TxType DefaultTxType = TxType.Legacy;
-        private static readonly Type[] _types = new Type[Transaction.MaxTxType + 1];
+        private static readonly List<TxTypeInfo> _txTypes = [];
```

Comment:
Why not allocate static, never growing array with: 
`private static readonly TxTypeInfo[] _types = new TxTypeInfo[Transaction.MaxTxType + 1];`
So we will have tx type mapping from index

Reply:
Because index there does not mean tx type any more, we check tx fields of tx type in ascending order. Such list will take less place also

Reply:
I actually think the tx type order is what is important not inserting order!
We can trim the array to smaller list if we don't want excessive items.